lib/rucio/client/downloadclient.py

```diff
@@ -1270,6 +1270,8 @@ def _prepare_items_for_download(self, did_to_options, merged_items_with_sources,
                 if options is None:
                     continue
                 file_item['merged_options'] = options
+                file_item.setdefault('transfer_timeout', options.get('transfer_timeout'))
```

Comment:
if I see correctly, options could be `None` here, right? will this still work then?

Reply:
No there is the `if options is None: continue` directly above. Actually options should never be `None` except there is a bug in the input merging.
Anyhow I would prefer to change the `_download_item` function to access the transfer_timeout and ignore_checksum fields correctly e.g. `item.get('merged_options', {}).get('transfer_timeout')` instead of copying the values. Could you change this please @nikmagini ?

Reply:
Ping @nikmagini

Reply:
> Anyhow I would prefer to change the `_download_item` function to access the transfer_timeout and ignore_checksum fields correctly e.g. `item.get('merged_options', {}).get('transfer_timeout')` instead of copying the values. Could you change this please @nikmagini ?

I'll look into it - however, IIRC I didn't do that because the `options` dict isn't always copied to `merged_options` (e.g. for pfns download). So I think that I'll have to look both in `options` and in `merged_options` for these keys... I'll let you know

Reply:
You can just change it in the pfn download aswell

Reply:
```
options = item.setdefault('merged_options', {})
options.setdefault('ignore_checksum', item.pop('ignore_checksum', True))
options.setdefault('transfer_timeout', item.pop('transfer_timeout', None))
```

Reply:
Hi, read the suggestion too late - for now I changed it to check for the keys in both dicts.
I agree it would be better to make all methods consistent though.
Can look into it tomorrow...

Reply:
Did you look into it? :D

Reply:
Right now - finally pushed the new update...